Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1065,9 +1082,13 @@ extension SwiftLanguageServer {
   }
 
   private func fullDocumentDiagnosticReport(
-    _ req: DocumentDiagnosticsRequest
+    _ req: DocumentDiagnosticsRequest,
+    useCache: Bool = false
   ) async throws -> RelatedFullDocumentDiagnosticReport {
     let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    if useCache, let report = report(for: snapshot.id) {
+      return report
```

Comment:
i was trying to add a test for the code changes, but i didn't find a way to do it.
at least this change doesn't break any existing tests.